server/scram/cryptoUtils.ts

```diff
@@ -0,0 +1,58 @@
+"use strict";
+import * as CryptoJS from "crypto-js";
+
+export function saltAndHashPassword(password: string, salt: string, iterationCount: number): string {
+    return CryptoJS.PBKDF2(
+        password,
+        CryptoJS.enc.Hex.parse(salt),
+        {
+            keySize: 160 / 32,
+            iterations: iterationCount,
+            hasher: CryptoJS.algo.SHA1,
+        }
+    ).toString(CryptoJS.enc.Hex);
+}
```

Comment:
I think it would be better to create a single TS function which, given `password: string, salt: string, iterationCount: number`, would return a json with `serverKey` and `storedKey`, i.e.
`{
        "storedKey": storedKey,
        "serverKey": serverKey
 }`